src/bls12_381/fr.rs

```diff
@@ -16,6 +17,12 @@ const R: FrRepr = FrRepr([0x1fffffffe, 0x5884b7fa00034802, 0x998c4fefecbc4ff5, 0
 // R2 = R^2 % r
 const R2: FrRepr = FrRepr([0xc999e990f3f29c6d, 0x2b6cedcb87925c23, 0x5d314967254398f, 0x748d9d99f59ff11]);
 
+// G1 has cofactor  (x - 1)^2 / 3  = 76329603384216526031706109802092473003
+pub const G1_COFACTOR: Fr = Fr(FrRepr([0x31207728cdb90779, 0xffea90400320679a, 0xa479130bcc248091, 0x6d41ecdf98f9d8cc]));
+
+// G2 has cofactor (x^8 - 4x^7 + 5x^6 - 4x^4 + 6x^3 - 4x^2 - 4x + 13) / 9.
+pub const G2_COFACTOR: Fr = Fr(FrRepr([0x755c12a6851eb90e, 0xe117af10228fcb8c, 0xd67937ea756c9ced, 0x6dabbfa8c0fd834b]));
```

Comment:
For each group, assert that the cofactor times the group order is the curve order.